trin-core/src/jsonrpc.rs

```diff
@@ -220,7 +220,15 @@ fn handle_request(
         })
         .to_string()),
         _ if obj.method.as_str().starts_with("discv5") => dispatch_portal_request(obj, portal_tx),
-        _ => dispatch_infura_request(obj, infura_url),
+        _ => match infura_url {
+            Some(url) => dispatch_infura_request(obj, &url),
+            None => Err(json!({
+                "jsonrpc": "2.0",
+                "id": obj.id,
+                "error": "Infura project id not provided!".to_string(),
```

Comment:
Could be nice to include the environment variable name in the error message so folks know how to resolve the problem.